peps/pep-9999.rst

```diff
@@ -157,36 +157,406 @@ splitting off headers and other build-time dependencies in a separate
+  can register their name mapping(s).
+- A standardized schemsa that defines how mappings should be structured.
+- The above documents should be written in a structured, human-readable file format.
+  As long as it can be validated with the corresponding JSON Schemas, they could be
+  delivered as JSON, YAML, TOML and/or others. Authors recommend JSON or YAML.
+- One central Python package for the central registry, list of ecosystems and known
+  mappings.
+
+On the client side, there should be:
+
 - A way for the system to specify a default (e.g., the Python install on Ubuntu
   could register ``apt`` as the default system package manager with the
-  registry tool. It may also be left unspecified.
-- A way for the user to specify the default and/or current system package
-  manager. E.g., a user on Ubuntu may want either ``apt``, ``conda``, ``brew``
-  or ``spack`` as their package manager of choice to provide external
-  dependencies.
+  registry tool). It may also be left unspecified.
+- A way for the user to specify the default and/or current system package manager.
+  E.g., a user on Ubuntu may want either ``apt``, ``conda``, ``brew``  or ``spack``
+  as their package manager of choice to provide external dependencies.
 
 
 Specification
 =============
 
-
+Three schemas are proposed:
+
+1. A central registry of PURLs.
+2. A list of known ecosystems and their package manager names.
+3. The ecosystem-specific mappings of PURLs to package specifiers,
+   plus details of their package managers.
+
+The central registry defines which identifiers are recognized as canonical. Each entry MUST
+provide a valid PURL in the field ``id``, with an optional free form ``description`` text.
+Additionally some entries MAY refer to another entry via the ``provides`` field, which takes a
+list of strings already defined as ``id`` in the registry. This is useful for aliases
+(e.g. ``dep:generic/arrow`` and ``dep:github/apache/arrow``), and concrete implementations of
+a ``dep:virtual/`` entry (e.g. ``dep:generic/gcc`` would provide ``dep:virtual/compiler/c``).
+Entries without ``provides`` content or, if populated, only with ``dep:virtual/`` identifiers,
+are considered canonical.
+
+The list of known ecosystems assigns an identifier to each ecosystem and reports the
+canonical location for its mapping. The known package managers are also provided here
+so tools can map package manager names back to ecosystems without querying each mapping.
+
+.. JRG: Not sure if tying the package manager list to the known ecosystems list is worth it.
+
+The mappings specify which ecosystem-specific identifiers provide the canonical entries
+available in the central registry. Each mapping entry consists of an ``id`` field with
+the canonical identifier, plus an optional free form ``description`` text. The field ``specs``
+MUST contain a string, a list of a strings, or a dictionary that maps ``build``, ``host`` and
```

Comment:
This is hard to understand when reading it for the first time. The obvious question from the reader is "why all these things"? Can you add the explanation, perhaps with one or two examples if it helps? We can think about what should then stay here and what moves to the Rationale or Examples section.

Reply:
I rephrased a bit. We have examples later in the text, but only with links. Do you want me to add concrete excerpts here? I'll give it a try but we can also move them later.